core/src/main/scala/org/apache/spark/ErrorClassesJSONReader.scala

```diff
@@ -66,7 +66,6 @@ class ErrorClassesJsonReader(jsonFileURLs: Seq[URL]) {
     val errorInfo = errorInfoMap.getOrElse(
       mainErrorClass,
       throw SparkException.internalError(s"Cannot find main error class '$errorClass'"))
-    assert(errorInfo.subClass.isDefined == subErrorClass.isDefined)
```

Comment:
Remove assert for subClass, so we can create Exception without subClass. Like `MALFORMED_PROTOBUF_MESSAGE`